doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/gradle/ProcessDocTask.kt

```diff
@@ -0,0 +1,187 @@
+        .convention(10_000)
+
+    /**
+     * The processors to use. These must be fully qualified names, such as:
+     * `"com.example.plugin.MyProcessor"`
+     */
+    @get:Input
+    val processors: ListProperty<String> = factory
+        .listProperty(String::class.java)
+        .convention(emptyList())
+
+    /** The classpath of this task. */
+    @Classpath
+    val classpath: Configuration = project.maybeCreateRuntimeConfiguration()
+
+    /** Used by the task to execute [ProcessDocsGradleAction]. */
+    @get:Inject
+    abstract val workerExecutor: WorkerExecutor
+
+    private fun Project.maybeCreateRuntimeConfiguration(): Configuration =
+        project.configurations.maybeCreate("kotlinKdocIncludePluginRuntime") {
+            isCanBeConsumed = true
+            dependencies.add(project.dependencies.create("org.jetbrains.kotlin:kotlin-compiler:1.7.20"))
+            dependencies.add(project.dependencies.create("org.jetbrains.dokka:dokka-analysis:1.7.20")) // compileOnly in base plugin
+            dependencies.add(project.dependencies.create("org.jetbrains.dokka:dokka-base:1.7.20"))
+            dependencies.add(project.dependencies.create("org.jetbrains.dokka:dokka-core:1.7.20"))
+        }
+
+    private fun <T : Any> NamedDomainObjectContainer<T>.maybeCreate(name: String, configuration: T.() -> Unit): T =
+        findByName(name) ?: create(name, configuration)
+
+    /**
+     * Adds dependency to plugin.
+     * Don't forget to add any new processor to the [processors] list.
+     *
+     * @param dependencyNotation Dependency notation
+     *
+     * @see org.gradle.api.artifacts.dsl.DependencyHandler.create
+     */
+    fun addPlugins(vararg dependencyNotation: Any) {
+        classpath.dependencies.addAll(
+            dependencyNotation.map { project.dependencies.create(it) }
+        )
+    }
+
+    /**
+     * Adds dependency to plugin.
+     * Don't forget to add any new processor to the [processors] list.
+     *
+     * @param dependency Dependency
+     */
+    fun addPlugins(vararg dependency: Dependency) {
+        classpath.dependencies.addAll(dependency)
+    }
+
+    private fun println(message: Any?) {
+        if (debug.get()) {
+            kotlin.io.println(message)
+        }
+    }
+
+    init {
+        outputs.upToDateWhen { false }
+    }
+
+    @TaskAction
+    fun process() {
+        println("Hello from plugin 'nl.jolanrensen.docProcessor'")
+
+        val sourceRoots = sources.get()
+        val target = target.get()
+        val runtime = classpath.resolve()
+        val processors = processors.get()
+
+        val relativeSources = sourceRoots.map { it.relativeTo(baseDir.get()) }
+        (targets as ConfigurableFileCollection).setFrom(
+            relativeSources.map {
+                File(target, it.path)
+            }
+        )
+
+        if (target.exists()) target.deleteRecursively()
+        target.mkdir()
+
+        println("Using target folder: $target")
```

Comment:
Can we use logger instead all these println stuff?
For example, log4j + kotlin logger (you could copy the approach and config from the KotlinDL)